pyaerocom/stats/stats.py

```diff
@@ -0,0 +1,268 @@
+        - R_spearman (Spearman corr. coeff) -> R_spearman
+        - R_kendall (Kendall's tau) ->
+
+    Note
+    ----
+    Nans are removed from the input arrays, information about no. of removed
+    points can be inferred from keys `totnum` and `num_valid` in return dict.
+
+    Parameters
+    ----------
+    data : ndarray
+        array containing data, that is supposed to be compared with reference
+        data
+    ref_data : ndarray
+        array containing data, that is used to compare `data` array with
+    data_filters : Optional[list[DataFilter]]
+        list of filter functions applied to the data arrays before calculating stats.
+    statistics : dict[str, StatisticsCalculator]
+        mapping between statistics name and functions to calculate that statistics.
+    stats_filter : Optional[list[StatisticsFilter]]
+        list of filter functions applied to the stats dictionary before returning the dict.
+    lowlim : float
+        lower end of considered value range (e.g. if set 0, then all datapoints
+        where either ``data`` or ``ref_data`` is smaller than 0 are removed).
+        Deprecated. Use data_filters with FilterByLimit instead.
+    highlim : float
+        upper end of considered value range
+        Deprecated. Use data_filters with FilterByLimit instead.
+    min_num_valid : int
+        minimum number of valid measurements required to compute statistical
+        parameters. Stat will be returned as NaN if length(data) is below this threshold.
+    weights: ndarray
+        array containing weights if computing weighted statistics.
+    drop_stats: tuple
+        tuple which drops the provided statistics from computed json files.
+        For example, setting drop_stats = ("mb", "mab"), results in json files
+        in hm/ts with entries which do not contain the mean bias and mean
+        absolute bias, but the other statistics are preserved.
+        Deprecated. Use stats_filter with instance of DropStats instead.
+    Returns
+    -------
+    StatsDict
+        dictionary containing computed statistics. Return mapping is the same as
+        provided in `statistics`.
+
+    Raises
+    ------
+    ValueError
+        if either of the input arrays has dimension other than 1
+    ValueError
+        if length of data is different to length of ref_data.
+    ValueError
+        if length of weights is different to length of data (assuming weights != None)
+    """
+    data = np.asarray(data)
+    ref_data = np.asarray(ref_data)
+
+    # Validation
+    if not data.ndim == 1 or not ref_data.ndim == 1:
+        raise ValueError("Invalid input. Data arrays must be one dimensional")
+
+    if len(data) != len(ref_data):
+        raise ValueError("Length mismatch between data and ref_data.")
+
+    if weights is not None:
+        if len(weights) != len(data):
+            raise ValueError("Invalid input. Length of weights must match length of data.")
+
+    #### ----- BACKWARDS COMPATIBILITY ------
```

Comment:
The backwards compatibility thing is not totally clear to me. Which of your changes would introduce breaking changes? Could you please document (in this issue) what would else would needed to be done to the source code in other places to move to this newer style of computing the statistics?

Reply:
For example `drop_stats` being deprecated? What is the preferred way of setting this in a config file now?

Reply:
Yeah, this is confusing.

I didn't want to change both the tests and the implementation at the same time, to increase confidence in the new implementation, so I designed `calculate_statistics()` to be able to be called exactly the same as the old function. I plan to remove this logic which will simplify it a lot.

The logic of dropping stats would be implemented by passing a `FilterDropStats` instance to `stat_filters` (eg. `calculate_statistics(..., stat_filters=[FilterDropStats(["R", ...])`, though longer term it probably makes more sense to not calculate them in the first place by not including them in the `statistics` dict.